Dodge.js

```diff
@@ -0,0 +1,179 @@
+function DeleteEntity(id){
+delete(ents[id])
+}
+
+//blows up the grenade
+function Explode(id){
+  //spawns shrapnel traveling from where the grenade blew up
+  InitialShrapnel(ents[id].x,ents[id].y)
+  //delete the original projectile
+  DeleteEntity(id)
+}
+
+//runs 60 times a second updating each entitiy
+function Gameloop(entities, { touches }, )  {
+  let toDelete = []
+  for (let id in entities) {
+      entities[id].update()
+  }
+  
+return entities
+}
+
+//creates a square component
+function CreateBrick(props) {
+return(
+  //Having a dynamic call as in props.Image produces the error
+  //Cannot find module './assets/Brick.jpeg'; which doesn't seem right
+    <Image source = {require("./assets/Brick.jpeg")}
+      style = {{        
+      position: 'absolute',
+      width: props.width,
+      height: props.height,
+      left: props.x,
+      top: props.y,
+      backgroundColor: props.backgroundColor,}}/>
+
+);
+}
+/*
+//creates a square component
+function CreateBrick(props){
+  return (
+     <View style={styles.container}>
+      <Image
+        style={ INSERT STYLE HERE }
+        source={require('@expo/snack-static/react-native-logo.png')}
+      />
+    
+    <View
+      style={{
+        position: 'absolute',
+        width: props.width,
+        height: props.height,
+        left: props.x,
+        top: props.y,
+        backgroundColor: props.backgroundColor,
+      }}
+    />
+  );
+}
+*/
+//creates a circle component
+function CreateCircle(props){
+  return (
+
+    <Image
+      source = {require("./assets/Baseball.png")}
+      style={{
+        position: 'absolute',
+        borderRadius: props.borderRadius,
+        width: props.width,
+        height: props.height,
+        left: props.x,
+        top: props.y,
+        backgroundColor: props.backgroundColor,
+      }}
+    />
+  );
+}
+
+  //creates one shrapnel in each of the cardinal directions
+  function InitialShrapnel(x, y){
+          let id = generateUniqueId()
+          ents[id] =  new Shrapnel(x,y,1,0,id) 
+          setTimeout(() => { DeleteEntity(id) }, 2000)
+          id = generateUniqueId()
+          ents[id] = new Shrapnel(x,y,0,1,id)
+          setTimeout(() => { DeleteEntity(id) }, 2000)
+          id = generateUniqueId()
+          ents[id] = new Shrapnel(x,y,0,-1,id)
+          setTimeout(() => { DeleteEntity(id) }, 2000)
+          id = generateUniqueId()
+          ents[id] = new Shrapnel(x,y,-1,0,id)
+          setTimeout(() => { DeleteEntity(id) }, 2000)
+        }
+
+// Navigates to the Results screen after an amount of time
+// Specified by a call in the main function
+function TimeUp(navigation){
+  navigation.navigate("Results", {counter: counter})
+}
+
+//Deletes all entities
+function DeleteEntities(){
+  for(let id in ents){
+    delete(ents[id])
+    console.log("DeleteEntities ran")
+  }
+}
+
+export default function Dodge({ navigation }) {
+  //this code only runs when navigated to from the home page, why?
```

Comment:
This is a property of the NavigationStack component you use in App.js.
Basically, it creates each screen *once* (maybe at the start? maybe when it's first displayed?) and then just switches which one is shown to the user. Basically like different tabs.

Reply:
Hmm Do I need to include this in the DodgeBall class component (the dodging game)

Reply:
The direction I was thinking is more like, implement your own screen change:
Have a function in App.js called "goToHomeScreen()" which deletes the current dodgeball game, and changes the display to a home message.
Have another function called "startDodgeBall()" which creates a *new* dodgeball game, then changes the display to show the game.
Have another function called "goToScoreScreen()" which takes a score to display, deletes the current dodgeball game, and changes the display to a screen showing the score.

Rather than passing "navigation" to DodgeBallGame you can pass these functions directly, or maybe an "App" object that has these functions as methods. [Similar to how each entity is passed a "DodgeBallGame", so that it can call functions like "addEntity()" and "deleteEntity()"]